examples/ruby/greeter_server_with_interrupt.rb

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env ruby
+
+# Copyright 2015 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Sample gRPC server that implements the Greeter::Helloworld service.
+#
+# Usage: $ path/to/greeter_server.rb
+
+this_dir = File.expand_path(File.dirname(__FILE__))
+lib_dir = File.join(this_dir, 'lib')
+$LOAD_PATH.unshift(lib_dir) unless $LOAD_PATH.include?(lib_dir)
+
+require 'grpc'
+require 'helloworld_services_pb'
+
+# GreeterServer is simple server that implements the Helloworld Greeter server.
+class GreeterServer < Helloworld::Greeter::Service
+  # say_hello implements the SayHello rpc method.
+  def say_hello(hello_req, _unused_call)
+    Helloworld::HelloReply.new(message: "Hello #{hello_req.name}")
+  end
+end
+
+# main starts an RpcServer that receives requests to GreeterServer at the sample
+# server port.
+def main
+  s = GRPC::RpcServer.new
+  s.add_http2_port('0.0.0.0:50051', :this_port_is_insecure)
+  s.handle(GreeterServer)
+  # will register signal handlers for signals SIGHUP, SIGINT and SIGQUIT to 
+  # shut down server gracefully
+  s.run_till_terminated_or_interrupted([1, 'sigint', 'quit'])
```

Comment:
nit: rather than add this new `greeter_server_with_interrupt.rb` server, can we just change `greeter_server.rb` to use the new API? Also same for the rest of the example servers like `route_guide_server.rb`.

We could then just add a comment to above the method invocation in these example servers that we could also use `run_till_terminated`, but are using `run_till_terminated_or_interrupted` because we want to gracefully shutdown on these signals.... something along those lines.

Reply:
Done